packages/use-contractkit/__tests__/utils/ethereum.test.ts

```diff
@@ -0,0 +1,74 @@
+import Web3 from 'web3';
+
+import {
+  AddEthereumEventListener,
+  Ethereum,
+  EthereumRequest,
+} from '../../src/global';
+import {
+  getEthereum,
+  getInjectedEthereum,
+  isEthereumFromMetamask,
+  isEthereumPresent,
+} from '../../src/utils/ethereum';
+
+const jestEthereumOn: AddEthereumEventListener = jest.fn();
+const jestEthereumRequest: EthereumRequest = jest.fn();
+const jestEthereum = jest.fn(() => ({
+  on: jestEthereumOn,
+  request: jestEthereumRequest,
+  enable: () => Promise.resolve(),
+  isMetaMask: true,
+}))();
+
+const setEthereum = (implementation?: Ethereum) => {
```

Comment:
Not a big deal but since you abstracted the window checks could we just make a moduleNameMapper path for the ethereum utils to a mock file rather than mess with the globals inside the test?